app/helpers/storage_resource_helper/textual_summary.rb

```diff
@@ -36,14 +35,40 @@ def textual_logical_total
     {:label => _("Total Size"), :value => number_to_human_size(@record.logical_total, :precision => 2)}
   end
 
-  def textual_capabilities
-    {:label => _("Capabilities"), :value => parse_capabilities(@record.capabilities)}
+  def textual_group_capabilities
+    capabilities = @record.capabilities.map do |c|
+      [c["name"], c["value"]]
+    end
+
+    TextualMultilabel.new(
+      _("Capabilities"),
+      :labels => [_("Capability"), _("Value")],
+      :values => capabilities
+    )
+  end
+
+  def textual_group_storage_services
+    services = @record.storage_services.map do |s|
+      [textual_link(s), s.description]
+    end
+
+    TextualMultilabel.new(
+      _("Storage Services"),
+      :labels => [_("Storage Service"), _("Description")],
+      :values => services
+    )
   end
 
-  def parse_capabilities(capabilities)
-    caps = capabilities.map do |capability|
-      "#{capability['name']}: #{capability['value']}"
+  def textual_group_cloud_volumes
+    volumes = @record.cloud_volumes.map do |v|
+      storage_service = @record.ext_management_system.storage_services.find(v.storage_service_id)
```

Comment:
Similar question, can you just do `StorageService.find(v.storage_service_id)` or use `v.storage_service` ?

Reply:
btw, we notices that the `textual_link` don't work in the resulting tables.
it looks like a link (blue) but clicking it gets no response.
I suspect the map function somehow disables it..

opened a bug-ticket in our next sprint to try and fix this throughout our provider.